src/Record/Record.Model/RecordBuilder.cs

```diff
@@ -336,15 +337,40 @@ public Record Build()
 
         var metadataGraph = CreateMetadataGraph();
 
-        var contentGraphId = metadataGraph.CreateBlankNode();
-        metadataGraph.Assert(new Triple(new UriNode(_storage.Id), new UriNode(new Uri(Namespaces.Record.HasContent)), contentGraphId));
+        if (_storage.ContentGraphs.Count == 0 && _storage.Quads.Count == 0 && _storage.Triples.Count == 0 && _storage.RdfStrings.Count == 0)
+        {
+            var metaDataTs = new TripleStore();
+            metaDataTs.Add(metadataGraph);
+            return new Record(metaDataTs);
+        }
 
+        var contentGraphId = metadataGraph.CreateBlankNode();
         var contentGraph = CreateContentGraph(contentGraphId, metadataGraph);
+        var checksumTriples = CreateChecksumTriples(_storage.ContentGraphs.Append(contentGraph));
+        metadataGraph.Assert(checksumTriples.Append(new Triple(new UriNode(_storage.Id), new UriNode(new Uri(Namespaces.Record.HasContent)), contentGraphId)));
+
         var ts = CreateTripleStore(metadataGraph, contentGraph);
 
```

Comment:
Should we move the logic about the content graph(s) to `CreateContentGraph`?

Reply:
We first need to create the content graphs and then compute the checksum. The triples created that describe the checksum is placed on the metadata graph, not the content graph.